src/scripts/main.js

```diff
@@ -1,3 +1,256 @@
 'use strict';
 
-// write code here
+// 1. Implement table sorting by clicking on the title (in two directions).
+const table = document.querySelector('table');
+let sortDirection = true;
+
+function getTableData() {
+  return [...table.querySelectorAll('tbody tr')].map((row) => {
+    return [...row.children].map((cell) => cell.innerText);
+  });
+}
+
+function renderTableData(data) {
+  const tbody = table.querySelector('tbody');
+
+  tbody.innerHTML = '';
+
+  data.forEach((rowData) => {
+    const row = document.createElement('tr');
+
+    rowData.forEach((cellData) => {
+      const cell = document.createElement('td');
+
+      cell.innerText = cellData;
+      row.appendChild(cell);
+    });
+    tbody.appendChild(row);
+  });
+}
+
+function sortTable(columnIndex) {
+  const data = getTableData();
+
+  data.sort((rowA, rowB) => {
+    let cellA = rowA[columnIndex];
+    let cellB = rowB[columnIndex];
+
+    if (columnIndex === 3) {
+      cellA = parseInt(cellA);
+      cellB = parseInt(cellB);
+    }
+
+    if (columnIndex === 4) {
+      cellA = parseFloat(cellA.slice(1).split(',').join(''));
+      cellB = parseFloat(cellB.slice(1).split(',').join(''));
+    }
+
+    if (sortDirection) {
+      return cellA > cellB ? 1 : -1;
+    }
+
+    return cellA > cellB ? -1 : 1;
+  });
+
+  renderTableData(data);
+  sortDirection = !sortDirection;
+}
+
+table.querySelectorAll('th').forEach((header, index) => {
+  header.addEventListener('click', () => sortTable(index));
+});
+
+// 2. When user clicks on a row, it should become selected.
+let activeRow = null;
+
+table.querySelectorAll('tbody tr').forEach((row) => {
+  row.addEventListener('click', () => {
+    if (activeRow) {
+      activeRow.classList.remove('active');
+    }
+
+    row.classList.add('active');
+    activeRow = row;
+  });
+});
```

Comment:
Bug: The event listeners are added only once when the script is loaded. If new rows are added to the table afterwards, they won't have the click event listener. Consider adding the event listener each time a new row is added.